src/components/screens/timebank-view-all-screen.tsx

```diff
@@ -0,0 +1,290 @@
+      setDisplayedPersonsTotalTime(personsTimeTotals);
+    } catch (error) {
+      setError(`${error} ${strings.error.fetchFailedGeneral}`);
+    }
+
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    fetchPersonsAndPersonsTotalTime();
+  }, []);
+
+  /**
+   * Renders the search
+   */
+  const renderSearch = () => (
+    <Box sx={{ display: "flex", justifyContent: "center" }}>
+      <TextField
+        value={searchInput}
+        onChange={handleSearchInputChange}
+        placeholder={strings.timebank.searchPlaceholder}
+        variant="standard"
+        disabled={loading}
+        sx={{ width: "99%", padding: 1 }}
+        InputProps={{
+          startAdornment: (
+            <InputAdornment position="start">
+              <Search />
+            </InputAdornment>
+          )
+        }}
+      />
+    </Box>
+  );
+
+  /**
+   * Handle search input change
+   *
+   * @param event input change event
+   */
+  const handleSearchInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const newSearchInput = event.target.value;
+    setSearchInput(newSearchInput);
+
+    if (newSearchInput === "") {
+      setDisplayedPersonsTotalTime(personsWithTotalTime);
+      return;
+    }
+
+    const newDisplayedPersonsTotalTime = personsWithTotalTime.filter((personsWithTotalTime) =>
+      `${personsWithTotalTime.person.firstName} ${personsWithTotalTime.person.lastName}`
+        .toLowerCase()
+        .includes(newSearchInput.toLowerCase())
+    );
+
+    setDisplayedPersonsTotalTime(newDisplayedPersonsTotalTime);
+  };
+
+  /**
+   * Get balance color
+   *
+   * @param personTotalTime person total time object
+   * @returns color string
+   */
+  const getBalanceColor = (personTotalTime: PersonTotalTime) =>
+    personTotalTime.balance > 0 ? theme.palette.success.main : theme.palette.error.main;
+
+  return (
+    <>
+      <Card sx={{ marginBottom: 2 }}>{renderSearch()}</Card>
+      <Grid container spacing={2} textAlign={"center"} marginBottom={20}>
+        {loading ? (
+          <CircularProgress sx={{ margin: "auto", mt: "5%", mb: "5%" }} />
+        ) : (
+          displayedPersonsTotalTime.map((personWithTotalTime) => {
+            return (
+              <Grid item xs={12} sm={6} md={4} key={personWithTotalTime.person.id}>
+                <Card
+                  sx={{
+                    display: "flex",
+                    flexDirection: "column",
+                    alignItems: "center",
+                    height: "400px",
+                    "& .recharts-wrapper .recharts-surface": {
+                      overflow: "visible"
+                    }
+                  }}
+                >
+                  <Box sx={{ padding: "10px", marginBottom: "60px" }}>
+                    <Typography variant="h4">{`${personWithTotalTime.person.firstName} ${personWithTotalTime.person.lastName}`}</Typography>
+                    <Typography>
+                      {personWithTotalTime.person.startDate &&
```

Comment:
Perhaps we should have a string here saying start date:
Just so it's clear what the date is referring to. Could be a hover tooltip if you think it looks better

Reply:
Added a simple string on the left side of the start date "Start Date: "